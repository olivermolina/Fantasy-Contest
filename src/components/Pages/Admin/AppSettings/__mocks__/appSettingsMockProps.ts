export const appSettingsMockProps = {
  appSettings: {
    BONUS_CREDIT_FREE_ENTRY_EQUIVALENT: '25',
    STAKE_TYPE_FREE_ENTRY: 'ALL_IN,INSURED',
    NUMBER_OF_PLAYERS_FREE_ENTRY: '2,3,4',
    MAX_MATCH_DEPOSIT_AMOUNT: '100',
    DEPOSIT_AMOUNT_OPTIONS: '10,25,50,75,100,200',
    RELOAD_BONUS_TYPE: 'FLAT',
    RELOAD_BONUS_AMOUNT: '0',
    MAX_RETENTION_BONUS: '50',
    RETENTION_BONUS_WEEKLY_CHANCE: '1',
    RETENTION_BONUS_MATCH_MULTIPLIER: '1',
    REFERRAL_CREDIT_AMOUNT: '25',
    REFERRAL_CUSTOM_TEXT:
      'Refer a friend from the link or code in your profile and get $25 in bonus credit!!',
    MIN_BET_AMOUNT: '1',
    MAX_BET_AMOUNT: '50',
    MAX_MARKET_ODDS: '150',
    MIN_MARKET_ODDS: '-150',
    initial: false,
    CHALLENGE_PROMO_MESSAGE:
      'Pick 2-4 players. Predict if they will get MORE or LESS than their projection.',
    REPEAT_ENTRIES_LIMIT: '1',
  },
  onSubmit: () => console.log('onSubmit'),
};
