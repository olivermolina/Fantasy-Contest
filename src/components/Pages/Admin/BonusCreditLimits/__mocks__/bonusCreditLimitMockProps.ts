import { BetStakeType } from '@prisma/client';

export const bonusCreditLimitMockProps = {
  onSubmit: () => console.log('submit'),
  data: {
    signupFreeEntry: false,
    bonusCreditFreeEntryEquivalent: 50,
    bonusCreditLimits: [
      {
        contestCategoryId: '1',
        numberOfPicks: 2,
        id: '30387acd-660c-46f0-a9e6-af87c007e6fb',
        enabled: true,
        stakeTypeOptions: [BetStakeType.ALL_IN, BetStakeType.INSURED],
        bonusCreditFreeEntryEquivalent: 25,
      },
      {
        contestCategoryId: '2',
        numberOfPicks: 3,
        id: '316555e5-b447-468a-b57f-381bbf64919a',
        enabled: true,
        stakeTypeOptions: [BetStakeType.ALL_IN, BetStakeType.INSURED],
        bonusCreditFreeEntryEquivalent: 50,
      },
      {
        contestCategoryId: '3',
        numberOfPicks: 4,
        id: 'fc063fcd-1a25-46e2-9361-d947eec1f9a9',
        enabled: true,
        stakeTypeOptions: [BetStakeType.ALL_IN, BetStakeType.INSURED],
        bonusCreditFreeEntryEquivalent: 40,
      },
      {
        contestCategoryId: '4',
        numberOfPicks: 5,
        id: 'a2542740-e90f-4763-8a3a-f7afba68898c',
        enabled: true,
        stakeTypeOptions: [BetStakeType.ALL_IN, BetStakeType.INSURED],
        bonusCreditFreeEntryEquivalent: 20,
      },
    ],
  },
  openAddUsersFreeEntryConfirmDialog: () => console.log('open'),
};
