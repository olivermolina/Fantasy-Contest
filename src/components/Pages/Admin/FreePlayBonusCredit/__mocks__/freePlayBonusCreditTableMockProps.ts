import { PaymentStatusCode } from '~/constants/PaymentStatusCode';
import { TransactionType } from '@prisma/client';

export const freePlayBonusCreditTableMockProps = {
  onSubmitDeletePlayFreeCredit: () => console.log('delete'),
  data: [
    {
      id: '1',
      transactionDate: new Date(),
      amountBonus: 100,
      status: PaymentStatusCode.COMPLETE,
      type: TransactionType.CREDIT,
    },
    {
      id: '2',
      transactionDate: new Date(),
      amountBonus: 100,
      status: PaymentStatusCode.CANCELLED,
      type: TransactionType.DEBIT,
    },
  ],
  totalBonusCreditBalance: 0,
};
