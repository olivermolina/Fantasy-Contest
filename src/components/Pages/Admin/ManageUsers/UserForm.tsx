import React, { useEffect } from 'react';
import {
  Box,
  FormHelperText,
  IconButton,
  MenuItem,
  Select,
  Tab,
  Tabs,
  TextField,
} from '@mui/material';
import Button from '@mui/material/Button';
import { UserStatus } from '@prisma/client';
import { Controller, useForm } from 'react-hook-form';
import { zodResolver } from '@hookform/resolvers/zod';
import * as z from 'zod';
import { ManageUserRowModel } from './ManageUsers';
import { NEW_USER_ID } from '~/constants/NewUserId';
import CloseIcon from '@mui/icons-material/Close';
import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
import { MobileDatePicker } from '@mui/x-date-pickers/MobileDatePicker';
import dayjs, { Dayjs } from 'dayjs';
import FormControl from '@mui/material/FormControl';
import { UserFormValidationSchema } from '~/schemas/UserFormValidationSchema';

interface TabPanelProps {
  children?: React.ReactNode;
  index: number;
  value: number;
}

function TabPanel(props: TabPanelProps) {
  const { children, value, index, ...other } = props;

  return (
    <div
      role="tabpanel"
      hidden={value !== index}
      id={`simple-tabpanel-${index}`}
      aria-labelledby={`simple-tab-${index}`}
      {...other}
    >
      {value === index && <>{children}</>}
    </div>
  );
}

function a11yProps(index: number) {
  return {
    id: `user-tab-${index}`,
    'aria-controls': `user-tabpanel-${index}`,
  };
}

/**
 * Defines the form fields for user form.
 */
export type UserFormInputs = z.infer<typeof UserFormValidationSchema>;

/**
 * Represents the properties of `UserForm` component.
 */
interface Props {
  user: ManageUserRowModel;
  onSubmit: (data: UserFormInputs) => void;
  closeForm: () => void;
}

/**
 * Defines the `UserForm` component.
 */
export default function UserForm(props: Props) {
  const { user, closeForm } = props;
  const {
    register,
    handleSubmit,
    formState: { errors },
    control,
    reset,
    setValue,
  } = useForm<UserFormInputs>({
    resolver: zodResolver(UserFormValidationSchema),
  });

  const [tabValue, setTabValue] = React.useState(0);

  const handleChange = (event: React.SyntheticEvent, newValue: number) => {
    setTabValue(newValue);
  };

  const [DOBValue, setDOBValue] = React.useState<Dayjs | null>(
    dayjs(user?.DOB),
  );

  const handleChangeDOB = (newDOBValue: Dayjs | null) => {
    setDOBValue(newDOBValue);
    if (newDOBValue) {
      setValue('DOB', newDOBValue.toDate());
    }
  };

  /**
   * Resets the form fields when `user` prop changes.
   */
  useEffect(() => {
    reset({
      id: user.id,
      username: user.username || '',
      password: user.id === NEW_USER_ID ? '' : user.username || '',
      email: user.email,
      phone: user.phone?.toString(),
      status: user.status,
      firstname: user.firstname || '',
      lastname: user.lastname || '',
      address1: user.address1 || '',
      address2: user.address2 || '',
      city: user.city || '',
      state: user.state || '',
      postalCode: user.postalCode || '',
      DOB: user.DOB || dayjs().toDate(),
    });
  }, [user]);

  // Reset the tab value when there are errors
  useEffect(() => {
    if (Object.keys(errors).length > 0) {
      // If there are errors in the first tab, set the tab value to 0
      if (errors.phone || errors.DOB) {
        setTabValue(0);
      } else {
        setTabValue(1);
      }
    }
  }, [errors]);

  return (
    <form onSubmit={handleSubmit(props.onSubmit)}>
      <div
        className={
          'flex justify-between bg-selected p-4 text-xl text-white items-center'
        }
      >
        <p className={'font-bold'}>
          {user.id === NEW_USER_ID ? 'Add ' : 'Edit '} User
        </p>
        <IconButton sx={{ color: 'inherit', mt: -1 }} onClick={closeForm}>
          <CloseIcon />
        </IconButton>
      </div>
      <Box sx={{ borderBottom: 1, borderColor: 'divider' }}>
        <Tabs
          value={tabValue}
          onChange={handleChange}
          aria-label="User Tabs"
          variant="fullWidth"
        >
          <Tab label="Account" {...a11yProps(0)} />
          <Tab label="Address" {...a11yProps(1)} />
        </Tabs>
      </Box>
      <TabPanel value={tabValue} index={0}>
        <div className={'flex flex-col  divide-y divide-dashed justify-center'}>
          <div
            className={'grid grid-cols-3 items-center lg:grid-cols-6 p-4 gap-2'}
          >
            <span className={'font-semibold'}>Status</span>
            <div className={'col-span-2 lg:col-span-5'}>
              <Controller
                name="status"
                control={control}
                defaultValue={user.status}
                render={({ field }) => (
                  <FormControl
                    fullWidth
                    error={!!errors?.status}
                    size={'small'}
                  >
                    <Select size={'small'} fullWidth {...field}>
                      <MenuItem key={'empty-status'} value={undefined}>
                        <span className={'italic text-gray-400'}>Status</span>
                      </MenuItem>
                      {[
                        UserStatus.ACTIVE,
                        UserStatus.INACTIVE,
                        UserStatus.SUSPENDED,
                      ].map((userStatus) => (
                        <MenuItem key={userStatus} value={userStatus}>
                          {userStatus}
                        </MenuItem>
                      ))}
                    </Select>

                    {errors?.status?.message ? (
                      <FormHelperText>{errors?.status?.message}</FormHelperText>
                    ) : null}
                  </FormControl>
                )}
              />
            </div>
          </div>
          <div
            className={'grid grid-cols-3 items-center lg:grid-cols-6 p-4 gap-2'}
          >
            <span className={'font-semibold'}>ID</span>
            <div className={'col-span-2 lg:col-span-5'}>
              <TextField
                variant={'outlined'}
                size={'small'}
                fullWidth
                {...register('id')}
                disabled
              />
            </div>
          </div>
          <div
            className={'grid grid-cols-3 items-center lg:grid-cols-6 p-4 gap-2'}
          >
            <span className={'font-semibold'}>Email</span>
            <div className={'col-span-2 lg:col-span-5'}>
              <TextField
                variant={'outlined'}
                size={'small'}
                fullWidth
                {...register('email')}
                error={!!errors?.email}
                helperText={errors?.email?.message}
                disabled={user.id !== NEW_USER_ID}
              />
            </div>
          </div>
          <div
            className={'grid grid-cols-3 items-center lg:grid-cols-6 p-4 gap-2'}
          >
            <span className={'font-semibold'}>Username</span>
            <div className={'col-span-2 lg:col-span-5'}>
              <TextField
                variant={'outlined'}
                size={'small'}
                fullWidth
                {...register('username')}
                error={!!errors?.username}
                helperText={errors?.username?.message}
                disabled={user.id !== NEW_USER_ID}
              />
            </div>
          </div>

          <div
            className={'grid grid-cols-3 items-center lg:grid-cols-6 p-4 gap-2'}
          >
            <span className={'font-semibold'}>Password</span>
            <div className={'col-span-2 lg:col-span-5'}>
              <TextField
                type="password"
                variant={'outlined'}
                size={'small'}
                fullWidth
                {...register('password')}
                error={!!errors?.password}
                helperText={errors?.password?.message}
                disabled={user.id !== NEW_USER_ID}
              />
            </div>
          </div>
          <div
            className={'grid grid-cols-3 items-center lg:grid-cols-6 p-4 gap-2'}
          >
            <span className={'font-semibold'}>Phone</span>
            <div className={'col-span-2 lg:col-span-5'}>
              <TextField
                type="number"
                variant={'outlined'}
                size={'small'}
                fullWidth
                {...register('phone')}
                error={!!errors?.phone}
                helperText={errors?.phone?.message}
              />
            </div>
          </div>
          <div
            className={'grid grid-cols-3 items-center lg:grid-cols-6 p-4 gap-2'}
          >
            <span className={'font-semibold'}>DOB</span>
            <div className={'col-span-2 lg:col-span-5'}>
              <LocalizationProvider dateAdapter={AdapterDayjs}>
                <MobileDatePicker
                  inputFormat="MM/DD/YYYY"
                  renderInput={(params) => (
                    <TextField
                      {...params}
                      variant="outlined"
                      fullWidth
                      error={!!errors?.DOB}
                      helperText={errors?.DOB?.message}
                    />
                  )}
                  value={DOBValue}
                  onChange={handleChangeDOB}
                />
              </LocalizationProvider>
            </div>
          </div>
        </div>
      </TabPanel>
      <TabPanel value={tabValue} index={1}>
        <div className={'flex flex-col  divide-y divide-dashed justify-center'}>
          <div
            className={'grid grid-cols-3 items-center lg:grid-cols-6 p-4 gap-2'}
          >
            <span className={'font-semibold'}>First Name</span>
            <div className={'col-span-2 lg:col-span-5'}>
              <TextField
                variant={'outlined'}
                size={'small'}
                fullWidth
                {...register('firstname')}
                error={!!errors?.firstname}
                helperText={errors?.firstname?.message}
              />
            </div>
          </div>
          <div
            className={'grid grid-cols-3 items-center lg:grid-cols-6 p-4 gap-2'}
          >
            <span className={'font-semibold'}>Last Name</span>
            <div className={'col-span-2 lg:col-span-5'}>
              <TextField
                variant={'outlined'}
                size={'small'}
                fullWidth
                {...register('lastname')}
                error={!!errors?.lastname}
                helperText={errors?.lastname?.message}
              />
            </div>
          </div>
          <div
            className={'grid grid-cols-3 items-center lg:grid-cols-6 p-4 gap-2'}
          >
            <span className={'font-semibold'}>Address 1</span>
            <div className={'col-span-2 lg:col-span-5'}>
              <TextField
                variant={'outlined'}
                size={'small'}
                fullWidth
                {...register('address1')}
                error={!!errors?.address1}
                helperText={errors?.address1?.message}
              />
            </div>
          </div>

          <div
            className={'grid grid-cols-3 items-center lg:grid-cols-6 p-4 gap-2'}
          >
            <span className={'font-semibold'}>Address 2</span>
            <div className={'col-span-2 lg:col-span-5'}>
              <TextField
                variant={'outlined'}
                size={'small'}
                fullWidth
                {...register('address2')}
              />
            </div>
          </div>
          <div
            className={'grid grid-cols-3 items-center lg:grid-cols-6 p-4 gap-2'}
          >
            <span className={'font-semibold'}>City</span>
            <div className={'col-span-2 lg:col-span-5'}>
              <TextField
                variant={'outlined'}
                size={'small'}
                fullWidth
                {...register('city')}
                error={!!errors?.city}
                helperText={errors?.city?.message}
              />
            </div>
          </div>
          <div
            className={'grid grid-cols-3 items-center lg:grid-cols-6 p-4 gap-2'}
          >
            <span className={'font-semibold'}>State</span>
            <div className={'col-span-2 lg:col-span-5'}>
              <TextField
                variant={'outlined'}
                size={'small'}
                fullWidth
                {...register('state')}
                error={!!errors?.state}
                helperText={errors?.state?.message}
              />
            </div>
          </div>
          <div
            className={'grid grid-cols-3 items-center lg:grid-cols-6 p-4 gap-2'}
          >
            <span className={'font-semibold'}>Postal/Zip</span>
            <div className={'col-span-2 lg:col-span-5'}>
              <TextField
                variant={'outlined'}
                size={'small'}
                fullWidth
                {...register('postalCode')}
                error={!!errors?.postalCode}
                helperText={errors?.postalCode?.message}
              />
            </div>
          </div>
        </div>
      </TabPanel>

      <div className={'flex flex-row justify-between p-4'}>
        <Button variant={'outlined'} onClick={closeForm}>
          Cancel
        </Button>
        <Button variant={'contained'} type={'submit'}>
          Save
        </Button>
      </div>
    </form>
  );
}
