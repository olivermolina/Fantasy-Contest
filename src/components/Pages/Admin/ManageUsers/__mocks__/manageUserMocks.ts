import { UserStatus, UserType } from '@prisma/client';
import { NEW_USER_ID } from '~/constants/NewUserId';

export const subAdminUsersMock = [
  {
    id: 'subAdmin1',
    email: 'test2@test.test',
    username: 'subAdmin1',
    status: UserStatus.ACTIVE,
    type: UserType.SUB_ADMIN,
    phone: '123456789',
    subAdminId: '',
  },
  {
    id: 'subAdmin2',
    email: 'test2@test.test',
    username: 'subAdmin2',
    status: UserStatus.ACTIVE,
    type: UserType.SUB_ADMIN,
    phone: '123456789',
    subAdminId: '',
  },
  {
    id: 'subAdmin3',
    email: 'test2@test.test',
    username: 'subAdmin3',
    status: UserStatus.ACTIVE,
    type: UserType.SUB_ADMIN,
    phone: '123456789',
    subAdminId: '',
  },
  {
    id: 'subAdmin4',
    email: 'test2@test.test',
    username: 'subAdmin4',
    status: UserStatus.ACTIVE,
    type: UserType.SUB_ADMIN,
    phone: '123456789',
    subAdminId: '',
  },
];
export const manageUserMocks = [
  {
    id: '2',
    username: 'test2',
    email: 'test2@test.test',
    status: UserStatus.ACTIVE,
    type: UserType.AGENT,
    phone: '123456789',
    subAdminId: subAdminUsersMock[0]!.id,
    password: '',
  },
  {
    id: '1',
    username: 'test1',
    email: 'test1@test.test',
    status: UserStatus.ACTIVE,
    type: UserType.SUB_ADMIN,
    phone: '123456789',
    subAdminId: '',
    password: '',
  },
  {
    id: '3',
    username: 'test3',
    email: 'test3@test.test',
    status: UserStatus.ACTIVE,
    type: UserType.AGENT,
    phone: '123456789',
    subAdminId: subAdminUsersMock[1]!.id,
    password: '',
  },
  {
    id: '4',
    username: 'test4',
    email: 'test4@test.test',
    status: UserStatus.INACTIVE,
    type: UserType.AGENT,
    phone: '123456789',
    subAdminId: subAdminUsersMock[0]!.id,
    password: '',
  },
  {
    id: '5',
    username: 'test5',
    email: 'test5@test.test',
    status: UserStatus.ACTIVE,
    type: UserType.AGENT,
    phone: '123456789',
    subAdminId: subAdminUsersMock[1]!.id,
    password: '',
  },
  {
    id: '6',
    username: 'test6',
    email: 'test6@test.test',
    status: UserStatus.ACTIVE,
    type: UserType.AGENT,
    phone: '123456789',
    subAdminId: subAdminUsersMock[1]!.id,
    password: '',
  },
  {
    id: '7',
    username: 'test7',
    email: 'test7@test.test',
    status: UserStatus.ACTIVE,
    type: UserType.AGENT,
    phone: '123456789',
    subAdminId: subAdminUsersMock[1]!.id,
    password: '',
  },
];

export const newUserMock = {
  id: NEW_USER_ID,
  username: '',
  email: '',
  status: UserStatus.INACTIVE,
  type: UserType.AGENT,
  phone: '',
  subAdminId: '',
};
