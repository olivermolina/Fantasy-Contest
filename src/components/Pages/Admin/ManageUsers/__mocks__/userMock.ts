import { UserStatus } from '@prisma/client';
import dayjs from 'dayjs';

export const userMock = {
  id: '1',
  username: 'test',
  password: 'test',
  email: 'test',
  phone: '12345',
  status: UserStatus.ACTIVE,
  firstname: '',
  lastname: 'test',
  address1: 'test',
  address2: 'test',
  city: 'test',
  state: 'test',
  postalCode: 'test',
  DOB: dayjs().toDate(),
  referral: 'test',
};
