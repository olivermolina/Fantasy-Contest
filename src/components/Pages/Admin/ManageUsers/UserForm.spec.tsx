import { render } from '@testing-library/react';
import UserForm from './UserForm';
import { userMock } from './__mocks__/userMock';

describe('UserForm', () => {
  it('should render correctly', () => {
    const mockProps = {
      user: userMock,
      closeForm: jest.fn(),
      onSubmit: jest.fn(),
    };

    const { container } = render(<UserForm {...mockProps} />);
    expect(container).toMatchSnapshot();
  });
});
