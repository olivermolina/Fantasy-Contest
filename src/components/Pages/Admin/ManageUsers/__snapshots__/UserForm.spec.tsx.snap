// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserForm should render correctly 1`] = `
<div>
  <form>
    <div
      class="flex justify-between bg-selected p-4 text-xl text-white items-center"
    >
      <p
        class="font-bold"
      >
        Edit 
         User
      </p>
      <button
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1qxc83e-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="CloseIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      class="MuiBox-root css-1gsv261"
    >
      <div
        class="MuiTabs-root css-1ujnqem-MuiTabs-root"
      >
        <div
          class="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
          style="overflow: hidden; margin-bottom: 0px;"
        >
          <div
            aria-label="User Tabs"
            class="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
            role="tablist"
          >
            <button
              aria-controls="user-tabpanel-0"
              aria-selected="true"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary MuiTab-fullWidth Mui-selected css-1ujykiq-MuiButtonBase-root-MuiTab-root"
              id="user-tab-0"
              role="tab"
              tabindex="0"
              type="button"
            >
              Account
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-controls="user-tabpanel-1"
              aria-selected="false"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary MuiTab-fullWidth css-1ujykiq-MuiButtonBase-root-MuiTab-root"
              id="user-tab-1"
              role="tab"
              tabindex="-1"
              type="button"
            >
              Address
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <span
            class="MuiTabs-indicator css-1aquho2-MuiTabs-indicator"
            style="left: 0px; width: 0px;"
          />
        </div>
      </div>
    </div>
    <div
      aria-labelledby="simple-tab-0"
      id="simple-tabpanel-0"
      role="tabpanel"
    >
      <div
        class="flex flex-col  divide-y divide-dashed justify-center"
      >
        <div
          class="grid grid-cols-3 items-center lg:grid-cols-6 p-4 gap-2"
        >
          <span
            class="font-semibold"
          >
            Active
          </span>
          <div
            class="col-span-2 lg:col-span-5"
          >
            <label
              class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
            >
              <span
                class="MuiSwitch-root MuiSwitch-sizeMedium css-julti5-MuiSwitch-root"
              >
                <span
                  class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked Mui-checked css-5ryogn-MuiButtonBase-root-MuiSwitch-switchBase"
                >
                  <input
                    class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
                    name="status"
                    type="checkbox"
                    value="true"
                  />
                  <span
                    class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
                  />
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </span>
                <span
                  class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
                />
              </span>
              <span
                class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
              />
            </label>
          </div>
        </div>
        <div
          class="grid grid-cols-3 items-center lg:grid-cols-6 p-4 gap-2"
        >
          <span
            class="font-semibold"
          >
            ID
          </span>
          <div
            class="col-span-2 lg:col-span-5"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            >
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input Mui-disabled MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                  disabled=""
                  id=":r0:"
                  name="id"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-ihdtdm"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </div>
        <div
          class="grid grid-cols-3 items-center lg:grid-cols-6 p-4 gap-2"
        >
          <span
            class="font-semibold"
          >
            Email
          </span>
          <div
            class="col-span-2 lg:col-span-5"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            >
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input Mui-disabled MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                  disabled=""
                  id=":r1:"
                  name="email"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-ihdtdm"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </div>
        <div
          class="grid grid-cols-3 items-center lg:grid-cols-6 p-4 gap-2"
        >
          <span
            class="font-semibold"
          >
            Username
          </span>
          <div
            class="col-span-2 lg:col-span-5"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            >
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input Mui-disabled MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                  disabled=""
                  id=":r2:"
                  name="username"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-ihdtdm"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </div>
        <div
          class="grid grid-cols-3 items-center lg:grid-cols-6 p-4 gap-2"
        >
          <span
            class="font-semibold"
          >
            Password
          </span>
          <div
            class="col-span-2 lg:col-span-5"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            >
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input Mui-disabled MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                  disabled=""
                  id=":r3:"
                  name="password"
                  type="password"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-ihdtdm"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </div>
        <div
          class="grid grid-cols-3 items-center lg:grid-cols-6 p-4 gap-2"
        >
          <span
            class="font-semibold"
          >
            Phone
          </span>
          <div
            class="col-span-2 lg:col-span-5"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            >
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                  id=":r4:"
                  name="phone"
                  type="number"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-ihdtdm"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </div>
        <div
          class="grid grid-cols-3 items-center lg:grid-cols-6 p-4 gap-2"
        >
          <span
            class="font-semibold"
          >
            DOB
          </span>
          <div
            class="col-span-2 lg:col-span-5"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            >
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl Mui-readOnly css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  aria-label="Choose date, selected date is Jan 1, 2021"
                  aria-readonly="true"
                  class="MuiInputBase-input MuiOutlinedInput-input Mui-readOnly css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id=":r5:"
                  readonly=""
                  type="text"
                  value="01/01/2021"
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-ihdtdm"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      aria-labelledby="simple-tab-1"
      hidden=""
      id="simple-tabpanel-1"
      role="tabpanel"
    />
    <div
      class="flex flex-row justify-between p-4"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Cancel
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="submit"
      >
        Save
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </form>
</div>
`;
